src/main/java/org/optaplanner/springboottaskassigning/TaskAssigningSolverManagerController.java

```diff
@@ -16,42 +16,68 @@
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
 
 @RestController
-@RequestMapping("solvers")
+@RequestMapping("/tenants/{tenantId}/solver")
 public class TaskAssigningSolverManagerController {
 
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
     @Autowired
-    private SolverManager<TaskAssigningSolution> solverManager;
+    private TaskAssigningSolverManagerService solverManagerService;
 
-    @PostMapping("{tenantId}")
-    public void solve(@PathVariable Comparable<?> tenantId, @RequestBody TaskAssigningSolution planningProblem) {
-        solverManager.solve(tenantId, planningProblem);
+    @PostMapping
+    public void solve(@PathVariable Long tenantId, @RequestBody TaskAssigningSolution planningProblem) {
+        logger.debug("POST /tenants/{}/solver", tenantId);
+        solverManagerService.solve(tenantId, planningProblem);
     }
 
-    @GetMapping("{tenantId}/bestSolution")
-    public TaskAssigningSolution bestSolution(@PathVariable Comparable<?> tenantId) {
-        return solverManager.getBestSolution(tenantId);
+    @GetMapping("/bestSolution")
+    public TaskAssigningSolution bestSolution(@PathVariable Long tenantId) {
+        logger.debug("GET /tenants/{}/solver/bestSolution", tenantId);
+        try {
+            return solverManagerService.bestSolution(tenantId);
+        } catch (NoSuchElementException e) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND,
+                    "Tenant (" + tenantId + ") does not have a solver task submitted or solving has not started yet.", e);
+        }
     }
 
-    @GetMapping("{tenantId}/bestScore")
-    public Score bestScore(@PathVariable Comparable<?> tenantId) {
-        return solverManager.getBestScore(tenantId);
+    @GetMapping("/bestScore")
+    public Score bestScore(@PathVariable Long tenantId) {
+        logger.debug("GET /tenants/{}/solver/bestScore", tenantId);
+        try {
+            return solverManagerService.bestScore(tenantId);
+        } catch (NoSuchElementException e) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND,
+                    "Tenant (" + tenantId + ") does not have a solver task submitted or solving has not started yet.", e);
+        }
     }
 
-    @GetMapping("{tenantId}/solverStatus")
-    public SolverStatus solverStatus(@PathVariable Comparable<?> tenantId) {
-        return solverManager.getSolverStatus(tenantId);
+    @GetMapping("/status")
+    public SolverStatus solverStatus(@PathVariable Long tenantId) {
+        logger.debug("GET /tenants/{}/solver/status", tenantId);
+        try {
+            return solverManagerService.solverStatus(tenantId);
```

Comment:
getSolverStatus

Reply:
Is it really necessary to follow the getter/setter notation? For me, just "solverStatus" is natural enough. But if we already have this convention in other APIs, then for the sake of consistency, +1.